src/client/browser/idle-page/index.js

```diff
@@ -1,24 +1,26 @@
 import * as browser from '../';
+import StatusMarkupCreator from './status-markup-creator';
 import COMMAND from '../../../browser-connection/command';
 
 
-const CHECK_STATUS_INTERVAL = 1000;
+const CHECK_STATUS_DELAY = 1000;
 
-
-var interval = null;
+var statusMarkupCreator = null;
 
 function checkStatus (statusUrl) {
     browser
         .checkStatus(statusUrl, window.XMLHttpRequest)
         .then((cmd) => {
-            if (cmd !== COMMAND.idle)
-                window.clearInterval(interval);
+            if (cmd === COMMAND.idle)
+                window.setTimeout(() => checkStatus(statusUrl), CHECK_STATUS_DELAY);
+        })
+        .catch(()=> {
+            statusMarkupCreator.changeStatus(false);
         });
 }
 
 window.init = function (statusUrl) {
-    interval = window.setInterval(() => checkStatus(statusUrl), CHECK_STATUS_INTERVAL);
-
     checkStatus(statusUrl);
-};
 
+    statusMarkupCreator = new StatusMarkupCreator(true);
```

Comment:
What we discussed was the *Markup Creator*. If it's actually the *Indicator* - that's a different story. I feel like that's a technical question for you, guys,
If it was .NET (sorry), the first thing would be non-visual, while the second would probably be a visual control. I think this is not the same with JS... So, if you typically give "control-ish" names to classes that create markup - than yes, this is an *indicator*.